department-of-veterans-affairs/caseflow
# Add active status for users

References #11631

### Description
Add status to the users column which defaults existing users and newly created users to "active

### Acceptance Criteria
- [x] Migration doesn't fail
- [x] All users are active after migration
- [x] Scope properly selects all active users
- [x] Creating a user defaults the status to active
- [x] Updating a user's status sets timestamp of change

### Testing Plan
1. Ensure migration causes no issues
2. Ensure all users are active after migration
```ruby
> User.count.eql?(User.active.count)`
=> true
```
3. Ensure creating a user defaults the status to active
```ruby
> FactoryBot.create(:user, station_id: 101, css_id: "NEW_USER")
=> #<User:0x00007fa8933bcef8
 id: 128,
 created_at: Fri, 16 Aug 2019 21:46:57 UTC +00:00,
 css_id: "NEW_USER",
 efolder_documents_fetched_at: nil,
 email: nil,
 full_name: "Lauren Roth",
 last_login_at: nil,
 roles: nil,
 selected_regional_office: nil,
 station_id: "101",
 updated_at: Fri, 16 Aug 2019 21:46:57 UTC +00:00,
 status: "active",
 status_updated_at: nil>
```
4. Ensure creating a user with an invalid status fails
```ruby
> FactoryBot.create(:user, station_id: 101, css_id: "NEW_USER_2",  status: "invalid")
ArgumentError: 'invalid' is not a valid status
```
5. Ensure updating a user's status updates the `status_updated_at` timestamp to now
```ruby
> User.last.update!(status: "inactive")
UPDATE "users" SET "status" = $1, "updated_at" = $2 WHERE "users"."id" = $3  [["status", "active"], ["updated_at", "2019-08-16 21:51:09.953541"], ["id", 128]]
UPDATE "users" SET "updated_at" = $1, "status_updated_at" = $2 WHERE "users"."id" = $3  [["updated_at", "2019-08-16 21:51:09.959955"], ["status_updated_at", "2019-08-16 21:51:09.959373"], ["id", 128]]
```

*Schema Changes Only*
